### PR TITLE
Implementation of  MCG storage class

### PR DESCRIPTION
**Pull request by parikshithb**

### CODE REVIEW

File: tests/helpers.py [hunk 1]
```diff
@@ -374,6 +374,30 @@ def create_storage_class(
     return create_resource(**sc_data)
 
 
+def create_mcg_storageclass(reclaim_policy=constants.RECLAIM_POLICY_DELETE, sc_name=None):
```
Comment (ebenahar) [inline tests/helpers.py]: I think you can utilize the already existing create_storageclass() function we have in helpers.py and tp pass it the MCG SC yaml

File: tests/manage/mcg/conftest.py [hunk 2]
```diff
@@ -145,3 +150,38 @@ def awscli_pod(mcg_obj, created_pods):
     helpers.wait_for_resource_state(awscli_pod_obj, constants.STATUS_RUNNING)
     created_pods.append(awscli_pod_obj)
     return awscli_pod_obj
+
+
+@pytest.fixture()
+def mcg_storageclass(request):
```
Comment (ebenahar) [inline tests/manage/mcg/conftest.py]: the same, try to utilize the storage class factory fixture we have

File: tests/helpers.py (hunk 1 above)
Comment (parikshithb) [reply tests/helpers.py]: create_storageclass() in helpers has certain default parameters like secret_name, interface_name which is not required to create a mcg storage class.

File: tests/manage/mcg/conftest.py (hunk 2 above)
Comment (parikshithb) [reply tests/manage/mcg/conftest.py]: Similary, storage class factory calls ceph_pool_factory_class and secret_factory_class which is not required for mcg sc unlike rbd and cephfs sc.

**Comment by parikshithb** [comment]
@ebenahar True, will wait on the resolution of this bz, if there is default storage class for OBCs created then there is no need in creating a separate storageclass.

**Comment by ebenahar** [comment]
@parikshithb ACK, thanks!

**Comment by parikshithb** [comment]
There is no need for creating a separate storageclass for obc as it is created by default and only reclaim policy delete is supported. Closing this PR.

**Pull request closed without merging**